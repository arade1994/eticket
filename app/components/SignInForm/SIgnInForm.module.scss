.signInForm {
  max-width: 500px;
  margin: 40px auto;
  display: flex;
  flex-direction: column;
  gap: 20px;
  background: #ffffff;
  padding: 40px;
  border-radius: 8px;
  box-shadow: 0 4px 20px rgba(0, 0, 0, 0.1);

  & h1 {
    color: #006064;
    text-align: center;
    font-weight: 700;
    font-size: 2rem;
    margin-bottom: 10px;
  }

  & .subtitle {
    text-align: center;
    color: #666;
    margin-top: -10px;
    margin-bottom: 30px;
  }

  & .formGroup {
    position: relative;
    display: flex;
    flex-direction: column;

    & .formControl {
      height: 45px;
      font-size: 16px;
      border: 1px solid #ccc;
      box-shadow: 0px 0px 3px rgba(0, 0, 0, 0.1);
      padding: 0 15px;
      border-radius: 5px;
      transition: all 0.3s ease;
      width: 100%;
      box-sizing: border-box;

      &:focus {
        outline: none;
        border-color: #00838f;
        box-shadow: 0 0 5px rgba(0, 131, 143, 0.5);
      }
    }
  }

  & .passwordToggle {
    position: absolute;
    right: 15px;
    top: 50%;
    transform: translateY(-50%);
    cursor: pointer;
    color: #006064;
    font-size: 1.2rem;
    user-select: none;
  }

  & .submitBtn {
    justify-self: center;
    width: 120px;
    height: 45px;
    border-radius: 5px;
    background-color: #036481;
    color: #fff;
    font-weight: bold;
    font-size: 16px;
    cursor: pointer;
    border: none;
    transition: background-color 0.3s ease, transform 0.2s ease;
    margin: 0 auto;

    &:hover {
      background-color: #006064;
      transform: translateY(-2px);
    }
  }
}
