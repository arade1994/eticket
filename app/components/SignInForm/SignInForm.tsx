import { type FormEvent, useCallback, useState } from "react";
import { FaEye, FaEyeSlash } from "react-icons/fa";

import { useAuth } from "../../hooks/useAuth";

import classes from "./SIgnInForm.module.scss";

const SignInForm: React.FC = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [showPassword, setShowPassword] = useState(false);

  const { login } = useAuth();

  const handleSubmit = useCallback(
    async (event: FormEvent<HTMLFormElement>) => {
      event.preventDefault();

      await login({ email, password });
    },
    [email, password, login]
  );

  return (
    <form className={classes.signInForm} onSubmit={handleSubmit}>
      <h1>Sign In</h1>
      <p className={classes.subtitle}>Welcome back! Please sign in.</p>
      <div className={classes.formGroup}>
        <input
          className={classes.formControl}
          id="email"
          placeholder="Email Address"
          type="email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </div>
      <div className={classes.formGroup}>
        <input
          className={classes.formControl}
          id="password"
          placeholder="Password"
          type={showPassword ? "text" : "password"}
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <span
          className={classes.passwordToggle}
          onClick={() => setShowPassword(!showPassword)}
        >
          {showPassword ? <FaEyeSlash /> : <FaEye />}
        </span>
      </div>
      <button className={classes.submitBtn}>Sign In</button>
    </form>
  );
};

export default SignInForm;
