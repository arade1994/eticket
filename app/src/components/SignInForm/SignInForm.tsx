import { type FormEvent, useCallback, useState } from "react";
import { toast } from "react-toastify";

import { useRequest } from "../../hooks/useRequest";

import classes from "./SIgnInForm.module.scss";

const SignInForm: React.FC = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const { sendRequest, errors } = useRequest({
    url: "/api/users/signin",
    method: "post",
    body: { email, password },
    onSuccess: () => {
      toast.success("User successfully signed in");
      Router.push("/");
    },
  });

  const handleSubmit = useCallback(
    async (event: FormEvent<HTMLFormElement>) => {
      event.preventDefault();

      await sendRequest();
    },
    [sendRequest]
  );

  return (
    <form className={classes.signInForm} onSubmit={handleSubmit}>
      <h1>Sign In</h1>
      <div className={classes.formGroup}>
        <label className={classes.formLabel}>Email Address:</label>
        <input
          className={classes.formControl}
          id="email"
          type="email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </div>
      <div className={classes.formGroup}>
        <label className={classes.formLabel}>Password:</label>
        <input
          className={classes.formControl}
          id="password"
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
      </div>
      {errors}
      <button className={classes.submitBtn}>Sign In</button>
    </form>
  );
};

export default SignInForm;
