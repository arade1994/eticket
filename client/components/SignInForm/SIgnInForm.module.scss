.signInForm {
  width: 400px;
  margin: auto;
  grid-column: 2/3;
  display: flex;
  flex-direction: column;
  gap: 25px;

  & .formGroup {
    display: flex;
    flex-direction: column;

    & .formLabel {
      color: gray;
    }

    & .formControl {
      height: 25px;
      font-size: 14px;
      border: none;
      box-shadow: 0px 0px 3px gray;
    }
  }

  & .submitBtn {
    width: 120px;
    height: 35px;
    border-radius: 5px;
    background-color: #036481;
    color: #fff;
    font-weight: bold;
    font-size: 14px;
    cursor: pointer;
  }
}
