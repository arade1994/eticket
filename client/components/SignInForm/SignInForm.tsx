import { FormEvent, useCallback, useState } from "react";
import { useRequest } from "../../hooks/useRequest";
import { toast } from "react-toastify";
import Router from "next/router";
import classes from "./SIgnInForm.module.scss";

const SignIn: React.FC = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const { sendRequest, errors } = useRequest({
    url: "/api/users/signin",
    method: "post",
    body: { email, password },
    onSuccess: () => {
      toast.success("User successfully signed in");
      Router.push("/");
    },
  });

  const handleSubmit = useCallback(
    async (event: FormEvent<HTMLFormElement>) => {
      event.preventDefault();

      await sendRequest();
    },
    [sendRequest]
  );

  return (
    <form className={classes.signInForm} onSubmit={handleSubmit}>
      <h1>Sign In</h1>
      <div className={classes.formGroup}>
        <label className={classes.formLabel}>Email Address:</label>
        <input
          className={classes.formControl}
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </div>
      <div className={classes.formGroup}>
        <label className={classes.formLabel}>Password:</label>
        <input
          className={classes.formControl}
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
      </div>
      {errors}
      <button className={classes.submitBtn}>Sign In</button>
    </form>
  );
};

export default SignIn;
