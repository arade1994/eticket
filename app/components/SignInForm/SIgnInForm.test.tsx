import { describe, expect, test, vi } from "vitest";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import SignInForm from "./SignInForm";

const mockSendRequest = vi.fn();

vi.mock("../../hooks/useRequest", async () => {
  const actualModule = await vi.importActual("../../hooks/useRequest");

  return {
    ...actualModule,
    useRequest: vi.fn().mockImplementation(() => ({
      sendRequest: mockSendRequest,
      errors: [],
    })),
  };
});

describe("<SignInForm />", () => {
  test("it should render sign in form as expected", () => {
    const { baseElement } = render(<SignInForm />);

    expect(baseElement).toMatchSnapshot();
  });

  test("it should render input for email address", async () => {
    render(<SignInForm />);

    const emailInput = screen.getByTestId("email") as HTMLInputElement;
    expect(emailInput).toBeDefined();
    await userEvent.type(emailInput, "test@test.com");
    expect(emailInput.value).toEqual("test@test.com");
  });

  test("it should render input for password", async () => {
    render(<SignInForm />);

    const passwordInput = screen.getByTestId("password") as HTMLInputElement;
    expect(passwordInput).toBeDefined();
    await userEvent.type(passwordInput, "password");
    expect(passwordInput.value).toEqual("password");
  });

  test("it should render submit button", () => {
    render(<SignInForm />);

    const signInBtn = screen.getByRole("button");
    expect(signInBtn).toBeDefined();
    expect(signInBtn.textContent).toEqual("Sign In");
  });

  test("it should call send request method when data are populated correctly", async () => {
    render(<SignInForm />);

    const emailInput = screen.getByTestId("email") as HTMLInputElement;
    await userEvent.type(emailInput, "test@test.com");
    const passwordInput = screen.getByTestId("password") as HTMLInputElement;
    await userEvent.type(passwordInput, "password");
    const signInBtn = screen.getByRole("button");
    await userEvent.click(signInBtn);
    expect(mockSendRequest).toHaveBeenCalled();
  });
});
