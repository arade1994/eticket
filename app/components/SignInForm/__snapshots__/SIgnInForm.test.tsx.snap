// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SignInForm /> > it should render sign in form as expected 1`] = `
<body>
  <div>
    <form
      class="_signInForm_070d36"
    >
      <h1>
        Sign In
      </h1>
      <p
        class="_subtitle_070d36"
      >
        Welcome back! Please sign in.
      </p>
      <div
        class="_formGroup_070d36"
      >
        <input
          class="_formControl_070d36"
          id="email"
          placeholder="Email Address"
          type="email"
          value=""
        />
      </div>
      <div
        class="_formGroup_070d36"
      >
        <input
          class="_formControl_070d36"
          id="password"
          placeholder="Password"
          type="password"
          value=""
        />
        <span
          class="_passwordToggle_070d36"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
            />
          </svg>
        </span>
      </div>
      <button
        class="_submitBtn_070d36"
      >
        Sign In
      </button>
    </form>
  </div>
</body>
`;
